### PR TITLE
refactor: move injection data to `t.func`

### PR DESCRIPTION
**Pull request by Natoandro**
- [MET-682](https://linear.app/metatypedev/issue/MET-682/move-all-injection-data-to-tfunc)
  - [x] Move all injection data in `ObjectTypeData` (i.e. `t.func`)
- [MET-656](https://linear.app/metatypedev/issue/MET-656/sdk-improve-generated-titles-from-applyreduce)
  - [x] Translate reduce to injection specification on `t.func`
- [MET-94](https://linear.app/metatypedev/issue/MET-94/remove-runtime-field-from-typenode)
  - [x] Remove runtime field from `TypeNode` (#858)
- [MET-683](https://linear.app/metatypedev/issue/MET-683/move-runtime-related-type-configs-out-of-typenode)
  - [x] Move runtime-related configs to `MaterializerData` or `RuntimeData`
- Misc.
  - Enable random ports for the typegate (when `TG_PORT=0`); this will work with `meta dev` with embedded typegate if you set the `--gate` option with port `0`.

#### Migration notes

...

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Yohe-Am — APPROVED** [review]
Great work!

**Review by Yohe-Am — DISMISSED** [review]
Great work!

File: tools/tasks/build.ts [hunk 1]
```diff
@@ -27,12 +27,12 @@ export default {
...
     inherit: ["_rust", "_wasm"],
     vars: {
-      WASM_FILE: "target/wasm/release/typegraph_core.wasm",
+      WASM_FILE: "target/wasm/debug/typegraph_core.wasm",
     },
     async fn($) {
       const target = "wasm32-unknown-unknown";
-      await $`cargo build -p typegraph_core --target ${target} --release --target-dir target/wasm`;
-      await $`cp target/wasm/${target}/release/typegraph_core.wasm $WASM_FILE.tmp`;
+      await $`cargo build -p typegraph_core --target ${target} --target-dir target/wasm`;
```
[the remark this answers is not in the document]
Comment (Natoandro) [reply tools/tasks/build.ts:34]: I used it for debugging purpose. I'll reset...

File: src/typegate/src/engine/planner/args.ts [hunk 2]
```diff
@@ -13,21 +13,22 @@ import type {
...
-  Type,
-  type TypeNode,
-  type UnionNode,
+import type {
+  ListNode,
+  ObjectNode,
+  TypeNode,
+  UnionNode,
 } from "../../typegraph/type_node.ts";
+import { getVariantTypesIndexes, Type } from "../../typegraph/type_node.ts";
```
Comment (michael-0acf4) [inline src/typegate/src/engine/planner/args.ts:22]: indices?

**Review by michael-0acf4 — COMMENTED** [review]
That's a lot of changes! Good stuff.

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**